include/aspect/utilities.h

```diff
@@ -1232,6 +1232,19 @@ namespace aspect
                               const SymmetricTensor<2,dim> &dviscosities_dstrain_rate,
                               const double SPD_safety_factor);
 
+    /**
+     * Computes the current_edot_ii and current_stress, with the option for
+     * viscous and viscoelastic contribution.
+     */
+    template <int dim>
+    double compute_current_edot_ii (const unsigned int j,  // the volume fraction
+                                                       const std::vector<double> &composition,
+                                                       const double ref_strain_rate,
+                                                       const double min_strain_rate,
+                                                       const SymmetricTensor<2,dim> &strain_rate,
+                                                       bool use_elasticity,
+                                                       bool use_reference_strainrate);
```

Comment:
These two are input arguments, which you can indicate by marking them as `const` just like all of the other arguments.